File: calculator/src/main/java/com/turlir/calculator/member/Operator.java
package com.turlir.calculator.member;

import com.turlir.calculator.converter.Member;
import com.turlir.calculator.converter.Visual;

public abstract class Operator implements Member, Comparable<Operator> {

    private final int mPriority;
    private final Visual mVisual;

    protected Operator(String token, int priority) {
        mPriority = priority;
        mVisual = new OperatorPrint(token);
    }

    @Override
    public int compareTo(Operator o) {
        return mPriority - o.mPriority;
    }

    @Override
    public boolean operand() {
        return false;
    }

    @Override
    public Visual view() {
        return mVisual;
    }
}

File: calculator/src/main/kotlin/com/turlir/calculator/member/Operator.kt
package com.turlir.calculator.member

import com.turlir.calculator.converter.Member
import com.turlir.calculator.converter.Visual

abstract class Operator protected constructor(private val token: String, private val mPriority: Int)
    : Member, Comparable<Operator> {

    private val mVisual: Visual = OperatorPrint(token)

    override fun compareTo(other: Operator) = mPriority - other.mPriority

    override fun operand() = false

    override fun view() = mVisual

    override fun toString(): String = "Operator(mPriority=$mPriority, token=$token)"
}
